packages/backend-core/src/objectStore/objectStore.ts

```diff
@@ -524,3 +524,20 @@ export async function getReadStream(
   }
   return client.getObject(params).createReadStream()
 }
+
+/*
+Given a signed url like 'files/signed/tmp-files-attachments/app_123456/myfile.txt' extract
+the bucket and the path from it
+*/
+export function extractBucketAndPath(url: string): {
+  bucket: string
+  path: string
+} {
+  const baseUrl = url.split("?")[0]
```

Comment:
this should probably check that it starts with the constant `files/signed` - if it doesn't then there is nothing it can extract - just to avoid any weird scenarios.

Reply:
Likewise, I'd check for the total number of `/` characters as well. Really wanna make sure you're not operating on something random, because bugs like that end up being very hard to track down.

Reply:
sorted this hopefully. This change also had the nice side effect of being able to get rid of that nasty header code in the email function. @mike12345567